### DISCUSSION TITLE
A new theme: swarms

### DISCUSSION THREAD

**Discussion by eawmaas** [xenocrat/chyrp-lite#225] [Show and tell]
I am working on an application for Chyrp Lite for which I need more flexibility in the future to add extra blocks, so I used Topaz which is my favorite theme to make a grid-based theme called swarms.

This theme standard has the sidebar on the right side and fixed with an overflow which gives two scrollbars. These things can easily be changed if you want, I have included a readme.md to give directions on what to modify and where.

Basically there are two grids: 1. is the main blog-grid and 2. is an extra sidebar-grid. The side-bar grid adds the possibility to rearrange the sidebar-items so that you can choose in which order they appear, again see readme.md for instructions.

I used grid-template-areas for both grids to assign the locations to each block, in the future I will add more columns and rows for my application but for this theme that is not necessary, so this is the clean new theme from which I can build further on my application.

If someone wants to try and encounters a problem please let me know and I will have a look, I have tested it on three different installs including the development-release and it seems to work fine.

UPDATE: The flash-notice when logged-in or out was not in the grid, so I added it to the top and it will span the whole page-width if applicable.
UPDATE II: Sorry, had to make some more adjustments, `div#main` wass behaving in an unexpected way from 1000 px up to around 1180 px, I have no idea where this behaviour comes from, there's nothing in the stylesheet to be found. But I added one mediaquery to solve it.
UPDATE III: Silly me forgot to rename the locale files... (I must write a testscript for the next time), but all good now

And I uploaded a new zipfile (again)

[swarms.zip](https://github.com/xenocrat/chyrp-lite/files/13330914/swarms.zip)

**Comment by xenocrat** [comment]
Thank you for sharing. I'll take a look!

**Comment by eawmaas** [comment]
It might need some tweaking still, but the idea is there. The fixed sidebar causes a lot of trouble, better to leave that part out first.

I was hoping to get a start with a grid-based layout for Chyrp light and ofcourse I will update the theme when it's perfect.
At least I can now add extra blocks to the content and the sidebar whenever I want and place everything wherever I want.

It didn't require many changes to the base-theme (Topaz) by the way so I think it has potential to become a modifyable blogtheme for the future. I am building a genealogy-tool and I want to try and integrate it into chyrp lite in the future, that's why I needed to be sure that it can be done making Chyrp grid-based because I need many more blocks for that tool.

Have fun!

**Comment by thethinkingman** [comment]
Hi there. Do you have it running somewhere?
